cachi2/core/errors.py

```diff
@@ -21,9 +79,25 @@ class GoModError(Cachi2Error):
     error is intermittent. We don't really know, but we do at least log the stderr.
     """
 
+    notice = textwrap.dedent(
+        """
+        The cause of the failure could be:
+        - something is broken in Cachi2
+        - something is wrong with your Go module
+        - communication with an external service failed (please try again)
+        The output of the failing go command should provide more details, please check the logs.
+        """
+    ).strip()
 
-class UnsupportedFeature(Cachi2Error):
-    """Cachi2 doesn't support a feature the user requested.
+    def friendly_msg(self) -> str:
+        """Return the user-friendly representation of this error."""
+        return _friendly_error_msg(str(self), self.notice, None)
 
-    The requested feature might be valid, but Cachi2 doesn't implement it.
-    """
+
+def _friendly_error_msg(reason: str, solution: Optional[str], docs_link: Optional[str]) -> str:
```

Comment:
Nitpick: should be called `_format_error_output` or something similar.

Reply:
I like the ideia of having functions named with verbs, but I don't think we're following this strictly, and that's fine.

I think I was a little bothered with having `friendly_msg` and `_friendly_error_msg`, but given the size and scope of this module, it's probably fine and easy to grasp the idea. As I said, nitpicking, feel free to ignore it.

Reply:
Last niptick on this: should we have these helper type of functions defined in the beginning of the file, or in the end, as it is here?

Reply:
We don't have any sort of consistency with this, but if we were to try and keep it consistent, I would vote for: public method first, helper method immediately after. In this case, since it's a helper function used by all the classes, "immediately after" is at the end of the file in my mind

But I generally don't think it's worth trying to enforce a code style which isn't enforced automatically by linters

Reply:
FWIW I fixed the order in the EnvFormat class to match ^